### PR TITLE
Making the -g flag optional for `create_xml_repo`

### PR DESCRIPTION
**Pull request by itsbenweeks**
Instead of passing owners and possibly removing people from that important group, let's make the group flag optional.

### CODE REVIEW

**Comment by itsbenweeks** [comment]
Buhh!? Tests passed but travis still failed. What gives? :boom:

File: orcoursetrion/tests/test_cmd.py [hunk 1]
```diff
@@ -111,6 +111,24 @@ def test_cmd_create_xml_repo(self):
...
+            '-m', 'archlight', 'dreadnought',
+        ]
+        with mock.patch('sys.argv', args):
+            with mock.patch('orcoursetrion.cmd.actions') as mocked_actions:
+                execute()
+                self.assertTrue(mocked_actions.create_xml_repo.called)
+                mocked_actions.create_xml_repo.assert_called_with(
+                    self.TEST_COURSE,
+                    self.TEST_TERM,
+                    None,
```
Comment (itsbenweeks) [inline orcoursetrion/tests/test_cmd.py]: Not sure why coveralls doesn't thing that I'm hitting the new conditional I added with this test. What am I missing?

File: orcoursetrion/actions/github.py [hunk 2]
```diff
@@ -171,7 +171,12 @@ def create_xml_repo(course, term, team, members=None, description=None):
     github.add_team_repo(
         config.ORC_XML_ORG, repo_name, config.ORC_XML_DEPLOY_TEAM
     )
-    # Setup the passed in team
+
+    # Team matches repo_name if no team is passed.
+    if (team is None):
```
Comment (carsongee) [inline orcoursetrion/actions/github.py]: no parens please

File: orcoursetrion/tests/test_cmd.py (hunk 1 above)
Comment (carsongee) [inline orcoursetrion/tests/test_cmd.py]: We have branch coverage turned on, I think, so you need to cover the condition being None, and being not None, though I would have thought one of the other tests would have picked that up.  Locally, does it show up as yellow in the html coverage report?

File: orcoursetrion/actions/github.py [hunk 3]
```diff
@@ -171,7 +171,12 @@ def create_xml_repo(course, term, team, members=None, description=None):
     github.add_team_repo(
         config.ORC_XML_ORG, repo_name, config.ORC_XML_DEPLOY_TEAM
     )
-    # Setup the passed in team
+
```
Comment (carsongee) [inline orcoursetrion/actions/github.py]: You need to update the doc string for this function

**Comment by carsongee** [comment]
You also need a unit test for `actions.create_xml_repo` in addition to the command line test

File: orcoursetrion/cmd.py [hunk 4]
```diff
@@ -153,12 +153,13 @@ def execute():
         help='Term of the course (i.e. Spring_2015)'
     )
     create_xml_repo.add_argument(
-        '-g', '--team', type=str, required=True,
-        help='Name of team in organization that should have access'
+        '-g', '--team', type=str,
```
Comment (carsongee) [inline orcoursetrion/cmd.py]: you should set a default of `None`

File: orcoursetrion/tests/test_cmd.py (hunk 1 above)
Comment (itsbenweeks) [inline orcoursetrion/tests/test_cmd.py]: That is what I had thought as well. This is covered with a team immediately after it.

File: orcoursetrion/cmd.py (hunk 4 above)
Comment (itsbenweeks) [inline orcoursetrion/cmd.py]: This may be why it wasn't adding coverage in the test, too.

**Comment by itsbenweeks** [comment]
Derp. I'm meeting some issues with the test I added because of the fake API. I'll have to pound on that tomorrow.

**Comment by itsbenweeks** [comment]
:clap: The separate functions worked! Thank you for helping me debug, @carsongee!

**Comment by carsongee** [comment]
Awesome.  This is looking super great :rice_ball:

**Pull request merged**